app/controllers/DashboardController.scala

```diff
@@ -92,9 +92,34 @@ trait DashboardController extends FrontendController with GatekeeperAuthWrapper
       fetchApplicationDetails(appId) map (details => Ok(review(HandleUpliftForm.form, details)))
   }
 
-  def approvedApplicationPage: Action[AnyContent] = requiresRole(Role.APIGatekeeper) {
+  def approvedApplicationPage(appId: String): Action[AnyContent] = requiresRole(Role.APIGatekeeper) {
     implicit request => implicit hc =>
-      Future.successful(Ok(approved()))
+
+      def lastApproval(app: ApplicationWithHistory): StateHistory = {
+        app.history.filter(_.state == State.PENDING_REQUESTER_VERIFICATION)
+          .sortWith(StateHistory.ascendingDateForAppId)
+          .lastOption.getOrElse(throw new InconsistentDataState("pending requester verification state history item not found"))
+      }
+
+      def administrators(app: ApplicationWithHistory): Future[Seq[User]] = {
+        val emails: Set[String] = app.application.admins.map(_.emailAddress)
+        developerConnector.fetchByEmails(emails.toSeq)
+      }
+
+      def application(app: ApplicationResponse, approved: StateHistory, admins: Seq[User], submissionDetails: SubmissionDetails) = {
+        val verified = app.state.name == State.PRODUCTION
+        val details = applicationDetails(app, submissionDetails)
+
+        ApprovedApplication(details, admins, approved.actor.id, approved.changedAt, verified)
+      }
+
+      for {
+        app <- applicationConnector.fetchApplication(appId)
```

Comment:
This code requires the application with history, which is not returned by the new fetchApplicationDetails method. Also, this is not used anywhere else, so can't see any benefit of extracting it.